### ISSUE TITLE
False positive E0742 (Visibility is restricted to incorrect module) on feature gated modules 

### ISSUE THREAD

**Issue by neonaot** [intellij-rust/intellij-rust#9320]
Originally posted in https://youtrack.jetbrains.com/issue/CPP-30193

## Environment

* **IntelliJ Rust plugin version:** 0.4.178.4857-222-nightly
* **Rust toolchain version:** 1.65.0-nightly (b44197abb 2022-09-05) x86_64-apple-darwin
* **IDE name and version:** CLion 2022.2.1 (CL-222.3739.54)
* **Operating system:** macOS 12.5.1
* **Macro expansion engine:** new
* **Name resolution engine:** new
* **Additional experimental features:** org.rust.cargo.features.settings.gutter, org.rust.cargo.emulate.terminal, org.rust.cargo.evaluate.build.scripts, org.rust.macros.proc

## Problem description
<img width="586" alt="image" src="https://user-images.githubusercontent.com/49211026/188627550-ee42c53b-9b5b-4f1f-aabe-cea315c4abf2.png">

> Because navigation on crate::gui takes me to not_gui, I assume the issue is that analyzer ignores the feature gates, and assumes all features are disabled. It is a serious usability problem, when most of the files under gui feature are red, because it wrongly interprets visibility and complains about items not being in correct scope, or being private. Build passes OK, with gui feature, and without it.
> 
> My suggested resolution is to use either last build to provide features, or currently selected "Run/Debug Configuration".

## Steps to reproduce
lib.rs:
```rust
#[cfg(feature = "gui")]
pub mod gui;
#[cfg(not(feature = "gui"))]
pub mod not_gui;
#[cfg(not(feature = "gui"))]
pub use not_gui as gui;
```
gui.rs:
```rust
pub (in crate::gui) struct Test;
```
